platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -97,8 +98,11 @@ export function ViewportGridProvider({ children, service }) {
         while (viewports.length < numPanes) {
           viewports.push({});
         }
-        while (viewports.length > numPanes) {
-          viewports.pop();
+
+        if (!keepExtraViewports) {
+          while (viewports.length > numPanes) {
+            viewports.pop();
+          }
         }
 
```

Comment:
why do we want to keep them around? can you add a comment above?

Reply:
I have a comment in ToolbarLayoutSelector.tsx, I suppose I could add a similar one here too.